### PR TITLE
Spark: Update register table procedure test case for all the catalogs

### PR DESCRIPTION
**Pull request by ajantha-bhat**
PR #[5037](https://github.com/apache/iceberg/pull/5037) supports register tables for all (most of) the catalogs. 
There is no restriction in the register table procedure about catalog type. But the testcase has a restriction. 

Hence, update the testcase to test for all the catalogs.

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
@RussellSpitzer, @aokolnychyi

File: spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java [hunk 1]
```diff
@@ -55,10 +53,6 @@ public void dropTables() {
 
   @Test
   public void testRegisterTable() throws NoSuchTableException, ParseException {
```
Comment (RussellSpitzer) [inline spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java:55]: So we aren't actually adding any more tests here we are just removing this Assume clause?

Comment (ajantha-bhat) [reply spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java:55]: Yes. 
Already API level test is there for each catalog for register table.
Configuring other catalogs in this test case is hard (like Glue, REST). It is better to keep them as integration tests (may in in a follow up PR)

Now removing this check will make sure that this testcase runs for hadoop, hive, spark catalog.

Comment (RussellSpitzer) [reply spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java:55]: How do we Register a table in Hadoop? That doesn't sound possible

Comment (ajantha-bhat) [reply spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java:55]: `HadoopCatalog` also extends `BaseMetastoreCatalog`

So, when a register table is called. It makes a fresh commit to the Hadoop catalog too (file name will be as per hadoop catalog)
https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java#L84

Comment (RussellSpitzer) [reply spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRegisterTableProcedure.java:55]: OK Got it, Still not sure I think that was a good idea but it's already in.

**Review by RussellSpitzer — APPROVED** [review]
Not sure we should be removing

**Pull request merged**